packages/geoprocessing/scripts/base/datasources/precalcRasterDatasource.ts

```diff
@@ -148,7 +149,8 @@ export async function precalcRasterMetrics(
   // Creates metrics for categorical raster (histogram, count valid cells by class)
   if (datasource.measurementType === "categorical") {
     const metrics: Metric[] = [];
-    const histogram = (await getHistogram(raster)) as Histogram;
+    const projectedFeat = toRasterProjection(raster, geographyFeatureColl);
+    const histogram = (await getHistogram(raster, projectedFeat)) as Histogram;
```

Comment:
Can this switch over to calling `rasterMetrics` like the quantitative raster code above?  It looks like you added support for categorical?

Reply:
The way I edited `rasterMetrics` is to return metrics for a specific categorical raster class, `category` (targeted for cycling through metric group classes in the gp function), so right now there isn't a way to return all histogram values as metrics. I'd be open to adding that though if it's desired. I'd have to split the `rasterMetrics` input parameters to include both a boolean `categorical` and a classKey `category`, so if `categorical = true` and `category = undefined` it returns all histogram values as metrics. LMK your thoughts

Reply:
If looping is being done in the gp function, one for each class, that sounds like extra calls to .histogram() and thus extra raster traversals?  Similar to what you said, I would suggest a `categorical` boolean, and an optional array of categories to generate metrics for.  Undefined categories array implies all categories.

Also, I would assume in the gp functions there is a process of converting numeric category values to class names?  In `rasterMetrics` when making it multi-band aware, I added a `bandMetricProperty` and `bandMetricValues` props and it does the mapping in rasterStatsToMetrics.  For categorical we could also do this.  Have like a `categoryMetricProperty` and `categoryClassValues: string[]` prop and it converts category numeric ID's to the string based.

Reply:
I realize this may be outside the scope of this pull request.  I'm open to it just getting ticketed.